code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/BasicNotificationBuilderTest.kt

```diff
@@ -0,0 +1,141 @@
+        remindLaterDuration = 3600
+    }
+
+    @Test
+    fun `construct should not return null`() {
+        val notificationBuilder = BasicNotificationBuilder.construct(context, basicPushTemplate, trackerActivityClass, broadcastReceiverClass)
+        assertNotNull(notificationBuilder)
+    }
+
+    @Test
+    fun `construct should return a NotificationCompat Builder`() {
+        val notificationBuilder = BasicNotificationBuilder.construct(context, basicPushTemplate, trackerActivityClass, broadcastReceiverClass)
+        assertEquals(NotificationCompat.Builder::class.java, notificationBuilder.javaClass)
+    }
+
+    @Test
+    fun `fallbackToBasicNotification should return a NotificationCompat Builder`() {
+        val notificationBuilder = BasicNotificationBuilder.fallbackToBasicNotification(
+            context,
+            trackerActivityClass,
+            broadcastReceiverClass,
+            dataMap
+        )
+        assertEquals(NotificationCompat.Builder::class.java, notificationBuilder.javaClass)
+    }
+
+    @Test
+    fun `createRemindPendingIntent should return PendingIntent with correct extended data`() {
+
+        val pushTemplate = BasicPushTemplate(MapData(dataMap))
+
+        val pendingIntent = BasicNotificationBuilder.createRemindPendingIntent(
+            context,
+            broadcastReceiverClass,
+            channelId,
+            pushTemplate
+        )
+
+        assertNotNull(pendingIntent)
+        val intent = shadowOf(pendingIntent).savedIntent
+
+        intent.putExtra(PushTemplateConstants.PushPayloadKeys.REMIND_LATER_TEXT, remindLaterText)
+        assertEquals(remindLaterText, intent.getStringExtra(PushTemplateConstants.PushPayloadKeys.REMIND_LATER_TEXT))
+
+        intent.putExtra(PushTemplateConstants.PushPayloadKeys.REMIND_LATER_TIMESTAMP, remindLaterTimestamp)
+        assertEquals(remindLaterTimestamp, intent.getLongExtra(PushTemplateConstants.PushPayloadKeys.REMIND_LATER_TIMESTAMP, -1))
+
+        intent.putExtra(PushTemplateConstants.PushPayloadKeys.REMIND_LATER_DURATION, remindLaterDuration)
+        assertEquals(remindLaterDuration, intent.getIntExtra(PushTemplateConstants.PushPayloadKeys.REMIND_LATER_DURATION, -1))
+
+        intent.putExtra(PushTemplateConstants.PushPayloadKeys.ACTION_BUTTONS, actionButtonsString)
+        assertEquals(actionButtonsString, intent.getStringExtra(PushTemplateConstants.PushPayloadKeys.ACTION_BUTTONS))
+        assertEquals(channelId, intent.getStringExtra(PushTemplateConstants.PushPayloadKeys.CHANNEL_ID))
+    }
+
+    @Test
+    fun `createRemindPendingIntent should return null when broadcastReceiverClass is null`() {
+
+        val pushTemplate = BasicPushTemplate(MapData(dataMap))
+
+        val pendingIntent = BasicNotificationBuilder.createRemindPendingIntent(
+            context,
+            null,
+            channelId,
+            pushTemplate
+        )
+
+        assertNull(pendingIntent)
+    }
+}
+
+class DummyActivity : Activity() {
```

Comment:
is it possible to use a mocked Activity and mocked BroadcastReceiver rather than dummy objects?